### ISSUE TITLE
2.0.33 to 2.0.34 update - all feeds disappeared

### ISSUE THREAD

**Issue by atomGit** [zakius/Smart-RSS#70]
after installing 2.0.34 no feed content was displayed - the feed/left pane was ok (folders/feed sources intact) except the unread counts were absent, the headline/center pane was empty as was the right/content pane

i downgraded to .33 and had the same issue

i then exported the feeds, both smart and opml, cleared all ext. data, updated to .34 and imported the smart feed - got an error and nothing was imported - then i imported the opml file and that was ok - everything returned to normal

however in trying to reproduce the smart import error, i deleted all ext. data again and re-imported - this time there was no error, but again both headline and content panes were empty and the unread counts were missing

at this time i am unable to import feeds - it appears the import is successful (i can export feeds) but nothing is displayed in the headline and content panes - refreshing feeds also appears to work, but nothing is displayed

note that i disabled my custom css for troubleshooting

there are no browser console errors

**Comment by zakius** [comment]
okay, this is really weird as I've been importing both types multiple times, using persistent data across commits and even already updated extension on my stable profile and nothing like that happened even once, but I'll do my best to figure it out and in the meantime I'll stop distribution of 34

can you tell me what is the exact browser version you're using?

**Comment by atomGit** [comment]
stable 66.0.5 on linux

the smart import error i was trying to reproduce was something like 'undefined db' or something - i might suggest not putting a lot of time into this until i can troubleshoot more - it may be an issue on my end

**Comment by zakius** [comment]
okay, few questions then:

- do you have any extension managing WebAPIs installed? (it uses IndexedDB to store data as other types of storage have unclear size constrains)
- was it maybe in the private window?
- do you have some restrictive privacy settings applied?

even if it was something unrelated to my changes I'd like to figure it out and at least create some basic FAQ/Troubleshooting document

what's weird is that SMART import failed for you, it was actually broken for a few versions but I noticed it and fixed it (I believe) in the 34, will look into this again to be sure

**Comment by zakius** [comment]
just to keep you updated: I found one bug that prevented feeds from being updated (typo, to be honest I have no idea why static analysis haven't marked that) but now I'm double checking everything else I touched lately, I'm sorry for inconvenience

**Comment by atomGit** [comment]
> do you have any extension managing WebAPIs installed? (it uses IndexedDB to store data as other types of storage have unclear size constrains)

very likely

> was it maybe in the private window?

no

> do you have some restrictive privacy settings applied?

absolutely :) i use a modified ghacks user.js + my own, however i don't disable IDB storage and i allow JS and everything else for the extension scheme

so i tested in a clean profile, but just once, and couldn't reproduce the problem of content not displaying, but i think that was with .33 - so then i went back to me default profile and again couldn't reproduce the problem which is why i didn't test more (yet) and why i assumed my extensions and prefs weren't the cause - this was all done previously as described

there shouldn't be anything messing with IDB storage - i allow that with uMatrix and Site Bleacher shouldn't do anything with extensions - i don't use a cookie manager since bleacher does all that

i think i'll duplicate my profile and try to repeat what i did - like i suggested, i wouldn't be too worried until i test more - nice that you found a lil bug though :)

**Comment by zakius** [comment]
I'm absolutely positive that I messed up updates with this version so you don't have to dig further

Now I have just one more issue that I may not be able to solve quickly: importing SMART somehow triggers GC and when it finishes it's impossible to access `sources` collection, fortunately everything works just fine after reloading the extension so for now I'll just add this as a workaround

**Comment by zakius** [comment]
Okay, I just published new version on AMO, it should be available shortly

**Comment by atomGit** [comment]
great that you found a problem and i see you already pub'd a new version

since i already did some additional troubleshooting i'll just leave this here in case it's of any use...

.34
no feed content
clear all data
import smart: ok
feeds updating
no feed content
clear all data
import opml: ok
feeds updating
no feed content
clear all data
uninstall
del /browser-extension-data/smart-rss@mozilla.firefox
del /extensions/staged
reinstall .33 (since .34 no longer available)
import smart: Importing error: TypeError: this.db is undefined
import opml
all is ok

**Comment by zakius** [comment]
right, there was SMART import issue for a few versions and that one typo was corrected in 34, but few others remained unfortunately

thanks for your feedback, closing this now since it's resolved

**Comment by atomGit** [comment]
.35 seems to be ok on my default profile - yippy!

.35
all ok with existing feeds and refreshing them
remove all data
import smart
all good!

i'll let you close this ... and thanks !!!

**Comment by zakius** [comment]
there's one inconvenience left with this issue: multiple of my sources redownloaded removed articles, possibly due to cleanup performed when no new articles were returned on failed load, I'm sorry for this

**Comment by atomGit** [comment]
not a biggie, but yeah, i have 2,285 unread articles LOL

what you could do also is add an option to cap the number of articles downloaded per feed - probably not very useful for existing users, but might be good for 1st time installs??? - which raises another issue - even if auto-updating is disabled, feeds still auto-update when importing - i'll create a new issue if you like

**Comment by zakius** [comment]
I'm not sure about the cap since it's impossible to request smaller portion of data in most cases, if it's in the downloaded document it makes sense to parse and save it I think (and you rarely see sources storing a lot of articles in their feed).

Currently update occurs only after OPML: import since I had to disable it due to SMART issues, but I see your point, feel free to submit an issue
